### PR TITLE
Closes #1641 drop names in if else

### LINKED ISSUE #1641: General Issue: Drop names in if_else()
### Background Information

Per https://github.com/pharmaverse/admiral/pull/1638 , 

`if_else()` is being updated for the upcoming release of `dplyr` 1.1.0 scheduled for January 27th, 2023. 

To make `admiral` compatible, adopt the changes as mentioned in above PR. 

### Definition of Done

- Adopt changes per https://github.com/pharmaverse/admiral/pull/1638 
- Double check for impact in other functions
- Update `NEWS.md` for 0.9.1
- Update `DESCRIPTION` 
- Remember to follow [hot fix release strategy](https://pharmaverse.github.io/admiraldev/devel/articles/release_strategy.html#hot-fix-release)

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bundfussr — APPROVED** [review]
In the future we should update `patch` directly after merging into `main` (https://github.com/pharmaverse/admiraldev/issues/199). Seeing all the changes from the last release makes it hard to review the pull request.

**Comment by bms63** [comment]
Any objections for me to just merge `main` into `patch` right now?  @bundfussr @zdz2101 ?

**Comment by zdz2101** [comment]
I say go for it

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# admiral 0.9.1
+
+- Implement changes to `if_else()` from the release of `dplyr` version 1.1.0. (#1641)
```
Comment (bms63) [inline NEWS.md:3]: I think we should reference the admiral function that is impacted by this change in dplyr

**Comment by bms63** [comment]
Much easier to look at now!!

File: NEWS.md (hunk 1 above)
Comment (zdz2101) [reply NEWS.md]: Updated!

**Review by bms63 — APPROVED** [review]
Thanks!

**Review by sadchla-codes — APPROVED** [review]

**Pull request merged**